src/etc/inc/util.inc

```diff
@@ -1932,19 +1932,34 @@ function mac_format($clientmac) {
 	}
 }
 
-function resolve_retry($hostname, $retries = 5) {
-
-	if (is_ipaddr($hostname)) {
-		return $hostname;
-	}
+function resolve_retry($hostname, $protocol = 'inet', $retries = 5) {
 
 	for ($i = 0; $i < $retries; $i++) {
-		// FIXME: gethostbyname does not work for AAAA hostnames, boo, hiss
-		$ip = gethostbyname($hostname);
-
-		if ($ip && $ip != $hostname) {
-			/* success */
-			return $ip;
+		switch ($protocol) {
+			case 'inet': 
+				$checkproto = 'is_ipaddrv4';
+				$ipproto = 'ip';
+				$dnsproto = array('ip' => DNS_A);
+				break;
+			case 'inet6':
+				$checkproto = 'is_ipaddrv6';
+				$dnsproto = array('ipv6' => DNS_AAAA);
+				break;
+			case 'any':
+				$checkproto = 'is_ipaddr';
+				$dnsproto = array('ip' => DNS_A, 'ipv6' => DNS_AAAA);
+				break;
+			default:
```

Comment:
Pair the default case with `inet` to preserve the old behavior if something passes it a null/empty value, just in case. I know nothing else in the base system uses it, but I haven't checked packages.

Reply:
done